src/buffer/out/textBuffer.cpp

```diff
@@ -4,15 +4,80 @@
 #include "precomp.h"
 
 #include "textBuffer.hpp"
-#include "CharRow.hpp"
+
+#include <til/hash.h>
 
 #include "../renderer/base/renderer.hpp"
 #include "../types/inc/utils.hpp"
 #include "../types/inc/convert.hpp"
 #include "../../types/inc/Utf16Parser.hpp"
 #include "../../types/inc/GlyphWidth.hpp"
 
-#pragma hdrstop
+struct BufferAllocator

```

Comment:
nit: consider `namespace {` to make sure this is unique across compilation units.

Reply:
(I love the unnamed namespace)

Reply:
Does that still hold true? All function definitions inside a struct definition are implicitly inline and so they shouldn't collide, right? 🤔